File: leetcode 227.java
class Solution {
    public int calculate(String s) {
        if(s.length() == 0 || s == null) return 0;
        Stack<Integer> stack = new Stack<Integer>();
        int num = 0;
        char sign = '+';
        for(int i = 0; i < s.length(); i++) {
            if(Character.isDigit(s.charAt(i)))
                num = num * 10 + s.charAt(i) - '0';
            if(!Character.isDigit(s.charAt(i)) && s.charAt(i) != ' ' || i == s.length() - 1) {
                if(sign == '+') stack.push(num);
                if(sign == '-') stack.push(-num);
                if(sign == '*') stack.push(stack.pop() * num);
                if(sign == '/') stack.push(stack.pop() / num);
                sign = s.charAt(i);
                num = 0;
            }
        }
        int sum = 0;
        for(int i : stack)
            sum += i;
        return sum;
    }
}

File: Python/leetcode 227.py
class Solution:
    def calculate(self, s):
        """
        :type s: str
        :rtype: int
        """
        if s == None or len(s) == 0: return 0
        stack = []
        num, sign = 0, '+'
        for i in range(len(s)):
            if s[i].isdigit(): num = num * 10 + int(s[i])
            if ((not s[i].isdigit()) and s[i] != ' ') or i == len(s)-1:
                if sign == '+': stack.append(num)
                if sign == '-': stack.append(-num)
                if sign == '*': stack.append(stack.pop() * num)
                if sign == '/': stack.append(int(stack.pop() / num))
                sign = s[i]
                num = 0
        Sum = 0
        for num in stack:
            Sum += num
        return Sum
        